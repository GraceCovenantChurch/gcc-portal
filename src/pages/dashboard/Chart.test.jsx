import React from "react";
import { shallow } from "enzyme";
import Chart from "./Chart";
import renderer from "react-test-renderer";

describe("Chart component", () => {
  let component, tree, wrapper;

  beforeEach(() => {
    const data = [
      createData("00:00", 0),
      createData("03:00", 300),
      createData("06:00", 600),
      createData("09:00", 800),
      createData("12:00", 1500),
      createData("15:00", 2000),
      createData("18:00", 2400),
      createData("21:00", 2400),
      createData("24:00", undefined),
    ];

    // Generate Sales Data
    function createData(time, amount) {
      return { time, amount };
    }

    component = renderer.create(<Chart data={data} />, { createNodeMock });
    tree = component.toJSON();

    wrapper = shallow(<Chart data={data} />);
  });

  it("UNIT: should be defined", () => {
    expect(Chart).toBeDefined();
  });

  it("SNAPSHOT: should render correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
