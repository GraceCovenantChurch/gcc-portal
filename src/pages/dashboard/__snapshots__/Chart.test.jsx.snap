// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component SNAPSHOT: should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Today
  </h2>,
  <div
    className="recharts-responsive-container"
    id="undefined"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div />
  </div>,
]
`;
